GCMessengerSDKSample/GCMessengerSDKSample/AccountDetailsViewController.swift

```diff
@@ -38,26 +38,44 @@ class AccountDetailsViewController: UIViewController {
         
         loggingSwitch.setOn(UserDefaults.logging, animated: true)
     }
-
+    
     @IBAction func startChatButtonTapped(_ sender: UIButton) {
+        validateFields(shouldStartChat: true)
```

Comment:
instead of passing true/false to the method consider passing the actual code block (reference to function).